splash/tests/test_execute.py

```diff
@@ -187,6 +187,28 @@ def test_bad_content_type_func(self):
         self.assertIn("argument must be a string", resp.text)
 
 
+class ResultHeaderTest(BaseLuaRenderTest):
+    def test_result_header_set(self):
+        resp = self.request_lua("""
+        function main(splash)
+            splash:set_result_header("foo", "bar")
+            return "hi!"
+        end
+        """)
+        self.assertStatusCode(resp, 200)
+        self.assertIn("foo", resp.headers)
+        self.assertEqual(resp.headers.get("foo"), "bar")
+
+    def test_bad_result_header_set(self):
+        resp = self.request_lua("""
+        function main(splash)
+            splash:set_result_header({}, [])
+            return "hi!"
+        end
+        """)
+        self.assertStatusCode(resp, 400)
```

Comment:
`splash:set_result_header({}, [])` is not valid Lua - there is no `[]`, so you'll get 400 error because of a syntax error, not because of validation error.